### PR TITLE
Fix rename and delete behavior to throw in certain cases

### PR DESCRIPTION
**Pull request by chemikadze**
Original HDFS rename implementation [returns](https://github.com/apache/hadoop/blob/release-2.9.2-RC0/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/protocol/ClientProtocol.java#L496:L508):

> true if successful, or false if the old name does not exists or or if the new name already belongs to the namespace.

In other cases exception is thrown.

This patch adds re-throwing in cases of access and server errors.

Some implementations, such as Apache Hive, expect rename to throw:
- https://github.com/apache/hive/blob/branch-2.1/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L2717
- https://github.com/apache/hive/blob/branch-2.1/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L2738

### CODE REVIEW

**Comment by chemikadze** [comment]
I signed it!

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java [hunk 1]
```diff
@@ -906,13 +907,10 @@ public boolean rename(Path src, Path dst) throws IOException {
     try {
       getGcsFs().rename(srcPath, dstPath);
```
Comment (functicons) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java]: Explain what kind of errors.

Comment (medb) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java]: It's already mentioned in java doc

**Pull request merged**